test/common/run-tests

```diff
@@ -301,139 +315,135 @@ def detect_tests(test_files, image, opts):
     return (serial_tests, parallel_tests)
 
 
-def run(opts, image):
-    result = 0
-    jobs = 1 if opts.list else opts.jobs
-    start_time = time.time()
+def list_tests(opts):
+    test_files = glob.glob(os.path.join(opts.test_dir, "check-*"))
+    (serial_tests, parallel_tests) = detect_tests(test_files, "dummy", opts)
+    names = [t.command[-1] for t in serial_tests] + [t.command[-1] for t in parallel_tests]
```

Comment:
`serial_tests` and `parallel_tests` are both lists, so you can DRY a bit here.  Also: what you're building is conceptually a set, so:

`names = {t.command[-1] for t in serial_tests + parallel_tests}` or so